aws/efs.go

```diff
@@ -0,0 +1,156 @@
+package aws
+
+import (
+	"fmt"
+	"time"
+	"github.com/gruntwork-io/go-commons/errors"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/efs"
+	"github.com/gruntwork-io/go-commons/retry"
+	"github.com/gruntwork-io/cloud-nuke/config"
+	"github.com/gruntwork-io/cloud-nuke/logging"
+)
+
+func getAllEfsVolumes(session *session.Session, region string, excludeAfter time.Time, configObj config.Config) ([]*string, error) {
+	svc := efs.New(session)
+
+	result, err := svc.DescribeFileSystems(&efs.DescribeFileSystemsInput{})
+	if err != nil {
+		return nil, errors.WithStackTrace(err)
+	}
+
+	var efsVolumeIds []*string
+	for _, volume := range result.FileSystems {
+		if shouldIncludeEFSVolume(volume, excludeAfter, configObj) {
+			efsVolumeIds = append(efsVolumeIds, volume.FileSystemId)
+		}
+	}
+
+	return efsVolumeIds, nil
+}
+
+func shouldIncludeEFSVolume(volume *efs.FileSystemDescription, excludeAfter time.Time, configObj config.Config) bool {
+	if volume == nil {
+		return false
+	}
+
+	if excludeAfter.Before(aws.TimeValue(volume.CreationTime)) {
+		return false
+	}
+
+	name := aws.StringValue(volume.Name)
+
+	return config.ShouldInclude(
+		name, 
+		configObj.EFSInstances.IncludeRule.NamesRegExp,
+		configObj.EFSInstances.ExcludeRule.NamesRegExp,
+	)
+}
+
+// Waits until EFS volume has been successfully nuked
+func waitUntilEfsVolumeIsNuked(session *session.Session, efsVolumeId *string) error {
+	svc := efs.New(session)
+
+	if efsVolumeId == nil {
+		return fmt.Errorf("EFS volume ID is invalid")
+	}
+
+	// We need to delete all volume mount targets of this EFS volume before destroying the EFS volume itself.
+	err := nukeEfsVolumeMountTargets(session, efsVolumeId)
+	if err != nil {
+		logging.Logger.Errorf("[Failed] %s", err)
+	}
+
+	_, err = svc.DeleteFileSystem(&efs.DeleteFileSystemInput{
+		FileSystemId: efsVolumeId,
+	})
+	if err != nil {
```

Comment:
Not sure if execution should continue after error logging, since `DeleteFileSystem` failed maybe make sense to exit function and skip calling `DoWithRetry`